.test-defs/cmd/shoot-maintenance/main.go

```diff
@@ -196,27 +234,52 @@ func main() {
 		testLogger.Fatalf("Failed to set machine images for cloud provider: '%s'", err.Error())
 	}
 	// update Cloud Profile with integration test machineImage
-	_, err = shootGardenerTest.GardenClient.Garden().GardenV1beta1().CloudProfiles().Update(shootMaintenanceTest.CloudProfile)
+	_, err = shootMaintenanceTest.ShootGardenerTest.GardenClient.Garden().GardenV1beta1().CloudProfiles().Update(shootMaintenanceTest.CloudProfile)
 	if err != nil {
 		testLogger.Fatalf("Failed to update Cloud Profile with integration test machineImage '%v'", err)
```

Comment:
I think we should replace the `logger.Fatal` calls with something else maybe wrap them in a function and return errors that we can react on.
Because when we exit with `Fatal` (which is basically a `os.Exit(1)`) all deferred function wont be called (https://play.golang.org/p/I5S8F7MD1WI), so the environment won't be restored.